themes/finna2/js/finna-carousel-manager.js

```diff
@@ -0,0 +1,155 @@
+        classes.prev += 'up';
+        classes.next += 'down';
+        classes.arrows += 'vertical';
+        direction = 'ttb';
+        break;
+      case 'carousel':
+        direction = 'ltr';
+        classes.prev += 'left';
+        classes.next += 'right';
+        classes.arrows += 'horizontal';
+        break;
+      }
+      return {
+        direction,
+        classes
+      };
+    },
+    slidesToScroll: 'perMove',
+    scrollSpeed: 'speed',
+    i18n: (translations) => {
+      return {
+        i18n: {
+          prev: VuFind.translate(translations.prev || 'Carousel::Prev'),
+          next: VuFind.translate(translations.next || 'Carousel::Next'),
+          first: VuFind.translate(translations.first || 'Carousel::First'),
+          last: VuFind.translate(translations.last || 'Carousel::Last'),
+          slideX: VuFind.translate(translations.slide || 'Carousel::go_to_page'),
+          pageX: VuFind.translate(translations.page || 'Carousel::page_number'),
+          play: VuFind.translate(translations.play || 'Carousel::Start Autoplay'),
+          pause: VuFind.translate(translations.pause || 'Carousel::Stop Autoplay'),
+          select: VuFind.translate(translations.select || 'Carousel::Select Page'),
+          slideLabel: VuFind.translate(translations.label || 'Carousel::slide_label'),
+        }
+      };
+    }
+  };
+
+  /**
+   * Merge sub objects into target from source
+   *
+   * @param {Object} target To merge key/values to
+   * @param {Object} source To merge key/values from
+   *
+   * @returns {Object} Merged object
+   */
+  function deepMerge(target, source) {
+    if (typeof target === 'object' && typeof source === 'object') {
+      for (const key in source) {
+        if (typeof source[key] === 'object') {
+          if (!target[key]) {
+            Object.assign(target, { [key]: {} });
+          }
+          deepMerge(target[key], source[key]);
+        } else {
+          Object.assign(target, { [key]: source[key] });
+        }
+      }
+    }
+    return target;
+  }
+
+  /**
+   * Converts settings into compatible Splide settings
+   *
+   * @param {Object} settings 
+   */
+  function toSplideSettings(settings) {
+    let splidied = {
+      direction: 'ltr',
+      gap: 10,
+      type: 'loop'
+    };
+    for (const [key, value] of Object.entries(settings)) {
+      if (typeof settingNameMappings[key] !== 'undefined') {
+        const newKey = settingNameMappings[key];
+        if (typeof newKey === 'function') {
+          const functionResult = newKey(value);
+          splidied = deepMerge(splidied, functionResult);
+        } else {
+          splidied[newKey] = value;
+        }
+      }
+    }
+
+    return splidied;
+  }
+
+  /**
+   * Turn given elements into carousels
+   *
+   * @param {Array|NodeList} elements Elements to turn into a carousel
+   * @param {Object}         settings Old Finna settings for carousels
+   */
+  function create(elements, settings) {
```

Comment:
Olisko `createCarousel` hyvä nimi, niin olisi selkeämpää, mitä `finna.carouselManager.createCarousel([this], settings);` tekee?